splunk/tests/splunk_test_prep.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// BuildSplunk builds the Splunk image used for testing Splunk queries.
+func BuildSplunk() {
+	buildCmd := exec.Command("podman", "build", "-t", "test_splunk", "./splunk")
+
+	buildCmd.Stdout = os.Stdout
+	buildCmd.Stderr = os.Stderr
+	err := buildCmd.Run()
+	if err != nil {
+		log.Fatalf("Failed to build image: %v", err)
+	}
+	log.Println("Image build done successfully!")
+}
+
+// RunSplunk deploy the Splunk container used for testing Splunk queries.
+func RunSplunk() string {
+	cmd := exec.Command("podman", "run", "-d", "-e", "SPLUNK_START_ARGS=--accept-license", "-e", "SPLUNK_PASSWORD=Password", "-p", "8000:8000", "-p", "8089:8089", "localhost/test_splunk:latest")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Fatalf("Failed to start command: %v", err)
+	}
+	containerID := strings.TrimSpace(string(output))
+	return containerID
+}
+
+// VerifyServiceIsUp continuously monitors the Splunk container to
+// ensure it is up and ready for use in tests. It achieves this by repeatedly
+// checking the status endpoint of the Splunk API over a period of two minutes.
+func VerifyServiceIsUp() {
+	timeoutStart := time.Now().Unix()
+	splunkURL := "https://localhost:8089/services/server/status"
+	username := "admin"
```

Comment:
Sensitive Information such as username and password should not be used directly in our code.

Reply:
Still a draft, but Is it sensetive? those are the default credentials for the Splunk instance we're getting up just for the tests. you can find those details in `pre-run.sh` and `log_indexing.sh` as well

Reply:
Then could you move these to const?

Reply:
Yes, Will be moved eventually

Reply:
done